src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -2068,13 +2068,11 @@ public List<CertificateDTO> getEntitlementCertificates(@Verify(Consumer.class) S
         }
 
         Consumer consumer = consumerCurator.verifyAndLookupConsumer(consumerUuid);
-        ConsumerType ctype = this.consumerTypeCurator.getConsumerType(consumer);
 
         revokeOnGuestMigration(consumer);
         poolManager.regenerateDirtyEntitlements(consumer);
 
         Set<Long> serialSet = this.extractSerials(serials);
-
         List<CertificateDTO> returnCerts = new LinkedList<>();
         List<EntitlementCertificate> allCerts = entCertService.listForConsumer(consumer);
```

Comment:
We can use some of the streaming API to reduce the code footprint and only have the filtering logic expressed once:

```
Stream<EntitlementCertificate> certStream = this.entCertService.listForConsumer(consumer)
    .stream();

Certificate caCert = this.contentAccessManager.getCertificate(consumer);
if (caCert != null) {
    certStream = Stream.concat(certStream, Stream.of(caCert));
}

Set<Long> serialSet = this.extractSerials(serials);
if (!serialSet.isEmpty()) {
    certStream.filter(cert -> serialSet.contains(cert.getSerial().getId()));
}

return certStream.map(translator.getStreamMapper(EntitlementCertificate.class, CertificateDTO.class))
    .collect(Collectors.toList());
```

I dunno if that's any cleaner though. Some improvements could be to do the stream building in a utility function or to always add the ca cert output and the filter out nulls (safer in the event `listForConsumer` ever misbehaves).

Reply:
I like combining both the certificates into a single collection and then filtering them all in a utility method! I wasn't sure if you were thinking of having the filtering method as a static method in some utility class or something, but I couldn't find a class that seemed to make sense. So, I put the logic in a private method in the ConsumerResource class.

Reply:
The one thing we want to avoid is extraneous copies. The streaming methods provide a view of the base data, and `Stream.concat` gives us a lazy concatenated view of two streams. 

The location of the utility method is fine, since it's a one-off here, and is what I had in mind. However, I was meaning to combine the individual cert and the list of certs in the utility method, since that's the messiest part of this code-wise, rather than moving the filtering out.

I'll take a closer look at this tomorrow morning and see if we can take this one step further.

Reply:
Yeah, I figured we could just add the individual cert that is returned from contentAccessManager.getCertificate to the list of certificates for the consumer, and then create a stream from that total collection of certificates that way we only have one stream we need to filter.

I also added filtering for null values in filterCertificatesBySerialIds to protect against a null value being returned from contentAccessManager.getCertificate.

Reply:
Okay, this makes sense. I added a `collect(Collectors.toList())` to the certStream.map so that we can still return a List and remain passive.

Reply:
Okay, that's fine here, especially since the way this end point works is a bit wonky; but in the future we should look for opportunities to convert those to streams when we're operating on potentially medium to large lists. Also makes for a good chance to deal with the spec file a bit more.